src/main/java/org/folio/dew/service/BulkEditParseService.java

```diff
@@ -0,0 +1,184 @@
+
+  private static final int ADDRESS_ID = 0;
+  private static final int ADDRESS_COUNTRY_ID = 1;
+  private static final int ADDRESS_LINE_1 = 2;
+  private static final int ADDRESS_LINE_2 = 3;
+  private static final int ADDRESS_CITY = 4;
+  private static final int ADDRESS_REGION = 5;
+  private static final int ADDRESS_POSTAL_CODE = 6;
+  private static final int ADDRESS_PRIMARY_ADDRESS = 7;
+  private static final int ADDRESS_TYPE = 8;
+
+  private static final int CF_KEY_INDEX = 0;
+  private static final int CF_VALUE_INDEX = 1;
+
+  public User mapUserFormatToUser(UserFormat userFormat) {
+    User user = new User();
+    populateUserFields(user, userFormat);
+    return user;
+  }
+
+  private void populateUserFields(User user, UserFormat userFormat) {
+    user.setId(userFormat.getId());
+    user.setUsername(userFormat.getUserName());
+    user.setExternalSystemId(userFormat.getExternalSystemId());
+    user.setBarcode(userFormat.getBarcode());
+    user.setActive(getIsActive(userFormat));
+    user.setType(userFormat.getType());
+    user.setPatronGroup(getPatronGroupId(userFormat));
+    user.setDepartments(getUserDepartments(userFormat));
+    user.setProxyFor(getProxyFor(userFormat));
+    user.setPersonal(getUserPersonalInfo(userFormat));
+    user.setEnrollmentDate(getDate(userFormat.getEnrollmentDate()));
+    user.setExpirationDate(getDate(userFormat.getExpirationDate()));
+    user.setCreatedDate(getDate(userFormat.getCreatedDate()));
+    user.setUpdatedDate(getDate(userFormat.getUpdatedDate()));
+    user.setTags(getTags(userFormat));
+    user.setCustomFields(getCustomFields(userFormat));
+  }
+
+  private boolean getIsActive(UserFormat userFormat) {
+    String value = userFormat.getActive();
+    if (value.matches("true") || value.matches("false")) {
+      return Boolean.parseBoolean(value);
+    }
+    //TODO in MODBULKED-14 save error that filed has a wrong value instead of returning false
+    return false;
+  }
+
+  private String getPatronGroupId(UserFormat userFormat) {
+    if (isNotEmpty(userFormat.getPatronGroup())) {
+      UserGroupCollection userGroup = userReferenceService.getUserGroupByGroupName(userFormat.getPatronGroup());
+      return userGroup.getUsergroups().iterator().next().getId();
+    }
+    return null;
+  }
+
+  private List<String> getUserDepartments(UserFormat userFormat) {
+    String[] departmentNames = userFormat.getDepartments().split(ARRAY_DELIMITER);
+    return Arrays.stream(departmentNames).parallel()
+      .map(userReferenceService::getDepartmentByName)
+      .flatMap(departmentCollection -> departmentCollection.getDepartments().stream())
+      .map(Department::getId)
+      .collect(Collectors.toList());
+  }
+
+  private List<String> getProxyFor(UserFormat userFormat) {
+    String[] proxyUserNames = userFormat.getProxyFor().split(ARRAY_DELIMITER);
+    return Arrays.stream(proxyUserNames)
+      .parallel()
+      .map(userReferenceService::getUserByName)
```

Comment:
Empty strings should be excluded from processing

Reply:
Corrected